py_pdf_parser/visualise/sections.py

```diff
@@ -162,42 +171,38 @@ def __plot_edges(
         # lines, or it is a LineString which means we only need to draw one.
         if isinstance(merged_line, geometry.MultiLineString):
             for line in merged_line:
-                xs, ys = self.__simplify_outlines(line, all_elements)
+                xs, ys = self.__simplify_outlines(line)
                 self.__ax.plot(xs, ys, **kwargs)
                 kwargs.pop(
                     "label", None
                 )  # Only pass label once for single legend entry
         else:
-            xs, ys = self.__simplify_outlines(merged_line, all_elements)
+            xs, ys = self.__simplify_outlines(merged_line)
             self.__ax.plot(xs, ys, **kwargs)
 
-    def __plot_section(
-        self,
-        pv: Pyvoronoi,
-        all_segments: List,
-        all_elements: List["PDFElement"],
-        section: "Section",
-        page: "PDFPage",
-    ):
-        edges = pv.GetEdges()
-        vertices = pv.GetVertices()
-        cells = pv.GetCells()
+    def __plot_section(self, section: "Section"):
+        if self.pv is None or self.pv_segments is None:
+            pv, pv_segments = self.__get_voronoi()
+            self.pv = pv
+            self.pv_segments = pv_segments
```

Comment:
This can be in one line:

```suggestion
            self.pv, self.pv_segments = self.__get_voronoi()
```